src/files/coverageparser.ts

```diff
@@ -17,147 +17,293 @@ export class CoverageParser {
      * Extracts coverage sections of type xml and lcov
      * @param files array of coverage files in string format
      */
-    public async filesToSections(files: Map<string, string>): Promise<Map<string, Section>> {
-        let coverages = new Map<string, Section>();
+    public async filesToSections(
+        files: Map<string, string>
+    ): Promise<Map<string, Section>> {
+        const coverages = new Map<string, Section>();
 
         for (const file of files) {
             const fileName = file[0];
             const fileContent = file[1];
 
-            // file is an array
-            let coverage = new Map<string, Section>();
-
             // get coverage file type
             const coverageFile = new CoverageFile(fileContent);
             switch (coverageFile.type) {
                 case CoverageType.CLOVER:
-                    coverage = await this.xmlExtractClover(fileName, fileContent);
+                    await this.xmlExtractClover(
```

Comment:
[F] Each of these `extract` methods now mutate the original `coverages` Map.

Reply:
Gotcha that makes sense to allow for the multiple coverage files scenario.